rfc/rfc-0002-bindings.md

```diff
@@ -0,0 +1,97 @@
+
+As a binding is backed by a ConfigMap and Secret a mechanism is needed to specify which ConfigMap and Secret. [LocalObjectReferences](https://godoc.org/k8s.io/api/core/v1#LocalObjectReference) are commonly used to define the coordinates to an arbitrary resource. A binding reference consists of `metadata` LocalObjectReference to a ConfigMap and a `secret` LocalObjectReference to a Secret.
+
+Any resource that provides a binding should include the binding reference on its status, for example:
+
+```yaml
+name: provider
+...
+status:
+  ...
+  binding:
+    metadataRef:
+      name: provider-binding-metadata
+    secretRef:
+      name: provider-binding-secret
+```
+
+The provider may change the values within the metadata and secret as needed, but should avoid changing the keys. The CNB Binding spec defines some well known keys that should be defined in the metadata.
+
+A workload consuming the binding can use the provider's binding, For example:
+
+```yaml
+name: consumer
+...
+spec:
+  ...
+  bindings:
+  - name: my-binding
+    metadataRef:
+      name: provider-binding-metadata
+    secretRef:
+      name: provider-binding-secret
+```
+
+It is common to consume multiple bindings within a single workload. Each binding must have a unique name for the workload. The binding name is the directory the metadata and secret volumes are mounted within. While the name has no impact on the binding per se, workloads may use the name to convey semantic meaning about the intent of the binding (e.g. a primary database vs a replica).
+
+Note: The ConfigMap and Secret for the binding must be in the same namespace as the workload Pod.
+
+The reconciler for the resource consuming a binding is responsible for mapping the ConfigMap and Secret to volume mounts within the pod.
+
+kpack, as of v0.0.5, does not provide any mechanism for injecting volumes into builds, or any other means to satisfy the CNB Bindings spec. Further work will be required for proper integration of bindings at buildtime.
+
+### User Impact
+This RFC does not require users to take any action. They remain free to use any other mechanism provided by Kubernetes to discover and/or inject credentials into their workloads. riff resources like Deployers and Processors currently contain a PodSpec where custom volumes, environment variables and arguments may be defined. Individual riff resources may start to produce and/or consume bindings, which users may then consume.
```

Comment:
what is meant by "consume" from the user's perspective here? might be confusing since "produce" and "consume" are used earlier in this sentence with more direct relevance to the preceding description of the feature

Reply:
```suggestion
This RFC does not require users to take any action. They remain free to use any other mechanism provided by Kubernetes to discover and/or inject credentials into their workloads. riff resources like Deployers and Processors currently contain a PodSpec where custom volumes, environment variables and arguments may be defined. Individual riff resources may start to produce and/or consume bindings.
```